### ISSUE TITLE
Auto Correct Code Style

### ISSUE THREAD

**Issue by kartikeysaran** [ankidroid/Anki-Android#8915]
Auto correcting the Code Style in every pull request

**Comment by mikehardy** [comment]
Add git commit hook that runs formatter configured for our linter I think.
https://stackoverflow.com/a/14672883/9910298

**Comment by mikehardy** [comment]
here's a pre-packaged thing for it even https://github.com/justinludwig/gjfpc-hook

Note that the google java code style may differ from our style somewhat. We have struggled to integrate an automated formatter (or even checker) for our style in the past and to me the tooling is much more important than the actual style, because with proper tooling you can at least easily use and enforce *a* style, and the consistency is strongly preferred over any sort of niche preference for some particular style in my opinion.

In other words, I would easily convert to some other style (including entire codebase) if we had strong tooling support.

Curious if this handles kotlin though, we will need something similar for Kotlin as I don't want to do that yet but it is in the future. Looks like ktlint is the choice for that but I'm not big on Kotlin yet there may be a better choice https://github.com/pinterest/ktlint

**Comment by david-allison-1** [comment]
I believe the Kotlin migration will handle this
